File: web/src/store/deploy/deploy.ts
import {
  DatabaseDeploySpec,
  DeploySpec,
  ElasticsearchDeploySepc,
  RabbitmqDeploySpec,
  RedisDeploySpec,
  ECMDeploySpec,
  EnvCheckOption
} from './types'
import { Module } from 'vuex'
import { getEnvCheckRule } from '@/service/deploy/deploy'
import localCache from '@/utils/cache'

const defaultPathPrefix = '/home/edoc2/macrowing/edoc2v5/data/'

const deployModule: Module<DeploySpec, any> = {
  namespaced: true,
  state() {
    return {
      middleware_mode: 'standalone',
      redis: {
        redis_data_path: defaultPathPrefix + 'redis',
        redis_hosts: ''
      },
      rabbitmq: {
        rabbitmq_data_path: defaultPathPrefix + 'rabbitmq',
        rabbitmq_hosts: ''
      },
      elasticsearch: {
        elasticsearch_hosts: '',
        elasticsearch_data_path: defaultPathPrefix + 'es',
        elasticsearch_backup_path: defaultPathPrefix + 'es/backup'
      },
      database: {
        is_external: false,
        db_type: 'mysql',
        db_hosts: '',
        db_backup_path: defaultPathPrefix + 'dbbackup',
        db_data_path: defaultPathPrefix + 'mysql',
        db_user: '',
        db_pass: '',
        db_port: ''
      },
      app: {
        storage: {
          storage_type: '',
          storage_path: defaultPathPrefix + 'edoc2Docs',
          storage_url: '',
          storage_ak: '',
          storage_sk: '',
          storage_health_check: '',
          storage_bucket: ''
        },
        hosts: '',
        scheme: 'http',
        access_port: 80,
        docker0_network: '172.17.0.0/24',
        docker_gwbridge_network: '172.18.0.0/16',
        docker_overlay_macrowing_network: '10.1.0.0/24'
      }
    }
  },
  mutations: {
    changeDeploySpec(state, value: DeploySpec) {
      state = value
    },
    changeDeployMode(state, value: string) {
      state.middleware_mode = value
    },
    changeAppDeploySpec(state, value: ECMDeploySpec) {
      state.app = value
    },
    changeRedisDeploySpec(state, value: RedisDeploySpec) {
      state.redis = value
    },
    changeRabbitmqDeploySpec(state, value: RabbitmqDeploySpec) {
      state.rabbitmq = value
    },
    changeDatabaseDeploySpec(state, value: DatabaseDeploySpec) {
      state.database = value
    },
    changeElasticsearchDeploySpec(state, value: ElasticsearchDeploySepc) {
      state.elasticsearch = value
    }
  },
  actions: {},
  getters: {
    getDeploySpec(state) {
      return state
    },
    getRedisDeploySpec(state) {
      return state.redis
    },
    getDatabaseDeploySpec(state) {
      return state.database
    },
    getElasticsearchDeploySepc(state) {
      return state.elasticsearch
    },
    getRabbitmqDeploySpec(state) {
      return state.rabbitmq
    }
  }
}

export default deployModule

File: internal/goecmagent/model/v1/deploy.go
package v1

type DeploySpec struct {
	MiddlewareDeploySpec
	ECMDeploySpec `json:"app"`
}

type ECMDeploySpec struct {
	ECMStorageSpec                `json:"storage"`
	Hosts                         string `json:"hosts"`
	Scheme                        string `json:"scheme"`
	AccessPort                    int    `json:"access_port"`
	Docker0Network                string `json:"docker0_network"`
	DockerGwbridgeNetwork         string `json:"docker_gwbridge_network"`
	DockerOverlayMacrowingNetwork string `json:"docker_overlay_macrowing_network"`
}

type ECMStorageSpec struct {
	StorageType        string `json:"storage_type"`
	StoragePath        string `json:"storage_path,omitempty"`
	StorageUrl         string `json:"storage_url,omitempty"`
	StorageHealthCheck string `json:"storage_health_check,omitempty"`
	StorageAK          string `json:"storage_ak,omitempty"`
	StorageSK          string `json:"storage_sk,omitempty"`
	StorageBucket      string `json:"storage_bucket,omitempty"`
}

type MiddlewareDeploySpec struct {
	Mode          string                  `json:"middleware_mode"`
	Redis         RedisDeploySpec         `json:"redis"`
	Rabbitmq      RabbitmqDeploySpec      `json:"rabbitmq"`
	Database      DatabaseDeploySpec      `json:"database"`
	Elasticsearch ElasticsearchDeploySpec `json:"elasticsearch"`
}

type DatabaseDeploySpec struct {
	IsExternal   bool   `json:"is_external"`
	DBType       string `json:"db_type"`
	DBHosts      string `json:"db_hosts"`
	DBPort       string `json:"db_port,omitempty"`
	DBUser       string `json:"db_user,omitempty"`
	DBPass       string `json:"db_pass,omitempty"`
	DBDataPath   string `json:"db_data_path,omitempty"`
	DBBackupPath string `json:"db_backup_path,omitempty"`
}

type RedisDeploySpec struct {
	RedisHosts    string `json:"redis_hosts"`
	RedisDataPath string `json:"redis_data_path"`
}

type ElasticsearchDeploySpec struct {
	ElasticsearchHosts      string `json:"elasticsearch_hosts"`
	ElasticsearchDataPath   string `json:"elasticsearch_data_path"`
	ElasticsearchBackupPath string `json:"elasticsearch_backup_path,omitempty"`
}

type RabbitmqDeploySpec struct {
	RabbitmqHosts    string `json:"rabbitmq_hosts"`
	RabbitmqDataPath string `json:"rabbitmq_data_path"`
}

type SwarmJoin struct {
	SwarmID      string `json:"swarm_id,omitempty" form:"swarm_id" gorm:"column:swarm_id;comment:'swarm cluster id'"`
	MgrJoinToken string `json:"mgr_join_token" form:"mgr_join_token" gorm:"column:mgr_join_token;comment:'swarm manager join-token'"`
	WrkJoinToken string `json:"wrk_join_token" form:"wrk_join_token" gorm:"column:wrk_join_token;comment:'swarm worker join-token'"`
}
